pkg/dns/gcp/provider.go

```diff
@@ -34,10 +34,22 @@ type Config struct {
 	Project         string
 	UserAgent       string
 	CredentialsJSON []byte
+	Endpoints       []configv1.GCPServiceEndpoint
 }
 
 func New(config Config) (*Provider, error) {
-	dnsService, err := gdnsv1.NewService(context.TODO(), option.WithCredentialsJSON(config.CredentialsJSON), option.WithUserAgent(config.UserAgent))
+	options := []option.ClientOption{
+		option.WithCredentialsJSON(config.CredentialsJSON),
+		option.WithUserAgent(config.UserAgent),
+	}
+	for _, endpoint := range config.Endpoints {
+		if endpoint.Name == configv1.GCPServiceEndpointNameDNS {
+			options = append(options, option.WithEndpoint(endpoint.URL))
+			break
+		}
+	}
```

Comment:
Potentially you could wrap this for loop with the feature gate... I think that would be _technically_ correct

The installer will only lay down the config if the feature gate is enabled, so it is already _effectively_ gated. I would defer to the preferences of the owning team.